### PR TITLE
Nguyenluu/refactor layout

### PR DESCRIPTION
**Pull request by LuuTheNguyen**
#Effect CSS

### Description
  - https://devcoin.ml/kanboard/?controller=TaskViewController&action=show&project_id=1&task_id=1
### Preview
    Screenshot or GIF if available, you can use https://imgur.com/ for hosting

### Checklist
 - [X] Tested on Desktop
 - [ ] Tested on Tablet
 - [ ] Tested on Mobile
 - [X] Tested `Print mode`
 - [ ] Covered with test cases
 - [ ] Added proper label
 - [ ] Performed self-review

### CODE REVIEW

File: components/Layout/Layout.tsx [hunk 1]
```diff
@@ -1,12 +1,10 @@
 import Head from 'next/head'
-
-interface Props {
-    name: string
-}
+import { StyledContainer } from './style'
+import type { Props } from '.'
```
Comment (tietthinh) [inline components/Layout/Layout.tsx:3]: import from `interface`

File: components/Menu/index.ts [hunk 2]
```diff
@@ -1 +1,3 @@
 export { Menu } from './Menu'
+export * from './interface'
+export * from './styles'
```
Comment (tietthinh) [inline components/Menu/index.ts:3]: Why do we need to export styles ?

File: containers/Home/About/About.tsx [hunk 3]
```diff
@@ -1,11 +1,11 @@
-import { StyledAbout } from './styles'
+import { StyledAbout } from '.'
 
-import { AboutProps, Prop } from './interface'
+import type { Props } from '.'
```
Comment (tietthinh) [inline containers/Home/About/About.tsx:3]: Same here

File: containers/Home/About/index.ts [hunk 4]
```diff
@@ -1 +1,3 @@
 export { About } from './About'
+export * from './interface'
+export * from './styles'
```
Comment (tietthinh) [inline containers/Home/About/index.ts:3]: same here?

File: containers/Home/Achievement/Achievement.tsx [hunk 5]
```diff
@@ -1,16 +1,22 @@
 import { StyledContainerAchievement, StyledAchievement } from './styles'
+import type { Props } from '.'
+import { ultiCountDown } from '.'
```
Comment (tietthinh) [inline containers/Home/Achievement/Achievement.tsx:3]: same here

File: containers/Home/Achievement/index.ts [hunk 6]
```diff
@@ -1 +1,4 @@
 export { Achievement } from './Achievement'
+export * from './interface'
+export * from './styles'
+export * from './ulti'
```
Comment (tietthinh) [inline containers/Home/Achievement/index.ts:4]: same here?

`util` ?

File: containers/Home/Contact/Contact.tsx [hunk 7]
```diff
@@ -1,10 +1,10 @@
-import { StyledContact } from './styles'
-import { ContactProps, Prop } from './interface'
+import { StyledContact } from '.'
+import type { Props } from '.'
```
Comment (tietthinh) [inline containers/Home/Contact/Contact.tsx:2]: same here?

File: containers/Home/Contact/index.ts [hunk 8]
```diff
@@ -1 +1,3 @@
 export { Contact } from './Contact'
+export * from './interface'
+export * from './styles'
```
Comment (tietthinh) [inline containers/Home/Contact/index.ts:3]: same here?

File: containers/Home/Education/Education.tsx [hunk 9]
```diff
@@ -1,12 +1,12 @@
-import { StyledEducations, StyledEducationItem } from './styles'
-import { Prop } from './interface'
+import { StyledEducations, StyledEducationItem } from '.'
+import type { Props } from '.'
```
Comment (tietthinh) [inline containers/Home/Education/Education.tsx:2]: same here?

File: containers/Home/Education/index.ts [hunk 10]
```diff
@@ -1 +1,3 @@
 export { Education } from './Education'
+export * from './styles'
```
Comment (tietthinh) [inline containers/Home/Education/index.ts:2]: same here?

File: containers/Home/Home.tsx [hunk 11]
```diff
@@ -21,58 +13,22 @@ import {
     StyledSubTitleContent,
     StyledContainerService,
     StyledContainerEducation,
-} from './styles'
+    StyledWrapperCloseButtonOffCanvas,
+} from '.'
```
Comment (tietthinh) [inline containers/Home/Home.tsx:17]: same here?

File: containers/Home/Home.tsx [hunk 12]
```diff
@@ -21,58 +13,22 @@ import {
...
 
-import { About } from './About'
-import { Contact } from './Contact'
-import { Language } from './Language'
-import { Lib } from './Lib'
-import { Skill } from './Skill'
 import { Achievement } from './Achievement'
 import { Service } from './Service'
 import { Education } from './Education'
+import { SideBar } from '.'
```
Comment (tietthinh) [inline containers/Home/Home.tsx:24]: import full path if it's local

File: containers/Home/Language/Language.tsx [hunk 13]
```diff
@@ -3,20 +3,21 @@ import {
     StyledProcessRing,
     StyledWrapperProcessRing,
     StyledTitleProcessRing,
-} from './styles'
```
Comment (tietthinh) [inline containers/Home/Language/Language.tsx:6]: this one this correct

File: containers/Home/Language/ProcessRing.tsx [hunk 14]
```diff
@@ -1,37 +1,64 @@
...
 
-interface ProcessRingProp {
-    width: number
-    strokeWidth: number
-    percent?: number
-    className?: string
-    stroke: string
-}
+import type { ProcessRingProp } from '.'
+import { utilProcessCircle } from '.'
```
Comment (tietthinh) [inline containers/Home/Language/ProcessRing.tsx:5]: same here?

File: containers/Home/Language/index.ts [hunk 15]
```diff
@@ -1 +1,4 @@
 export { Language } from './Language'
+export * from './interface'
+export * from './styles'
```
Comment (tietthinh) [inline containers/Home/Language/index.ts:3]: same here?

File: containers/Home/Lib/Lib.tsx [hunk 16]
```diff
@@ -1,10 +1,10 @@
-import { StyledLib, StyledListLib } from './styles'
-import { LibProps, Prop } from './interface'
+import { StyledLib, StyledListLib } from '.'
+import type { Props } from '.'
```
Comment (tietthinh) [inline containers/Home/Lib/Lib.tsx:2]: same here?

File: containers/Home/Lib/ListLib.tsx [hunk 17]
```diff
@@ -1,8 +1,4 @@
-interface ListLibProp {
-    color?: string
-    text: string
-    className?: string
-}
+import type { ListLibProp } from '.'
```
Comment (tietthinh) [inline containers/Home/Lib/ListLib.tsx:1]: same here?

File: containers/Home/Lib/index.ts [hunk 18]
```diff
@@ -1 +1,3 @@
 export { Lib } from './Lib'
+export * from './styles'
```
Comment (tietthinh) [inline containers/Home/Lib/index.ts:2]: same here?

File: containers/Home/Service/Service.tsx [hunk 19]
```diff
@@ -6,14 +6,15 @@ import {
     StyledCompany,
     StyledHeaderSubTitleService,
     StyledHeaderTitleService,
-} from './styles'
+} from '.'
```
Comment (tietthinh) [inline containers/Home/Service/Service.tsx:9]: same here?

File: containers/Home/Service/Service.tsx [hunk 20]
```diff
@@ -6,14 +6,15 @@ import {
...
 
-import { ServiceProps, Prop } from './interface'
+import type { Props } from '.'
```
Comment (tietthinh) [inline containers/Home/Service/Service.tsx:11]: same here?

File: containers/Home/Service/index.ts [hunk 21]
```diff
@@ -1 +1,3 @@
 export { Service } from './Service'
+export * from './interface'
+export * from './styles'
```
Comment (tietthinh) [inline containers/Home/Service/index.ts:3]: same here?

File: containers/Home/SideBar.tsx [hunk 22]
```diff
@@ -0,0 +1,70 @@
+import {
+    StyledAboutHead,
+    StyledWrapperImage,
+    StyledIcon,
+    StyledName,
+    StyledDescription,
+    StyledAboutBody,
+    StyledDownloadCV,
+    StyledAboutFooter,
+} from '.'
```
Comment (tietthinh) [inline containers/Home/SideBar.tsx:10]: same here?

File: containers/Home/SideBar.tsx [hunk 23]
```diff
@@ -0,0 +1,70 @@
...
+
+import { About } from './About'
+import { Contact } from './Contact'
+import { Language } from './Language'
+import { Lib } from './Lib'
+import { Skill } from './Skill'
+import type { HeadProps } from '.'
```
Comment (tietthinh) [inline containers/Home/SideBar.tsx:17]: same here?

File: containers/Home/Skill/ProcessBar.tsx [hunk 24]
```diff
@@ -1,16 +1,26 @@
-import { ProcessBarProps } from './interface'
+import { useEffect, useState } from 'react'
+import type { ProcessBarProps } from '.'
```
Comment (tietthinh) [inline containers/Home/Skill/ProcessBar.tsx:2]: same here?

File: containers/Home/Skill/Skill.tsx [hunk 25]
```diff
@@ -1,11 +1,11 @@
-import { StyledSkill, StyledProcessBar } from './styles'
+import { StyledSkill, StyledProcessBar } from '.'
```
Comment (tietthinh) [inline containers/Home/Skill/Skill.tsx:1]: same here?

File: containers/Home/Skill/Skill.tsx [hunk 26]
```diff
@@ -1,11 +1,11 @@
...
 
-import { SkillProps, Prop } from './interface'
+import type { SkillProps, Props } from '.'
```
Comment (tietthinh) [inline containers/Home/Skill/Skill.tsx:3]: same here?

File: containers/Home/Skill/index.ts [hunk 27]
```diff
@@ -1 +1,3 @@
 export { Skill } from './Skill'
+export * from './styles'
```
Comment (tietthinh) [inline containers/Home/Skill/index.ts:2]: same here?

**Review by tietthinh — COMMENTED** [review]
Don't use import from '.' when it's in local or nested component/style

File: enum/common/Theme.ts [hunk 28]
```diff
@@ -0,0 +1,8 @@
+export enum ThemeEnum {
+    LIGHT = 'theme-light',
+    DARK = 'theme-dark',
+}
+
+export enum TransitionEnum {
+    THEME = '0.6s',
```
Comment (tietthinh) [inline enum/common/Theme.ts:7]: `DURATION` ? 🤔

**Review by tietthinh — APPROVED** [review]

File: containers/Home/Lib/ListLib.tsx [hunk 29]
```diff
@@ -1,8 +1,4 @@
-interface ListLibProp {
-    color?: string
-    text: string
-    className?: string
-}
+import type { ListLibProp } from './interface'
```
Comment (peace-bi) [inline containers/Home/Lib/ListLib.tsx:1]: Shoud put name has 's' suffix --> ListLibProps

File: containers/Home/Lib/interface.ts [hunk 30]
```diff
@@ -2,6 +2,12 @@ export interface LibProps {
     label: string
 }
 
-export interface Prop {
-    libs: LibProps[]
+export interface Props {
+    lib: LibProps[]
+}
+
+export interface ListLibProp {
```
Comment (peace-bi) [inline containers/Home/Lib/interface.ts:9]: should put name has 's' suffix --> ListLibProps

File: containers/Home/Skill/ProcessBar.tsx [hunk 31]
```diff
@@ -1,16 +1,26 @@
...
 
 export const ProcessBar: React.FC<ProcessBarProps> = ({
     className,
     label = '-',
     percent = 100,
     height = 10,
 }) => {
+    const [count, setCount] = useState(0)
+    useEffect(() => {
+        const countDown = setTimeout(() => {
```
Comment (peace-bi) [inline containers/Home/Skill/ProcessBar.tsx:12]: Can you use countdown hook which you created previous?

**Review by peace-bi — COMMENTED** [review]
Please view these review comment.
Thank you

**Review by peace-bi — APPROVED** [review]
Review oke.

**Pull request merged**